org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/extension/AbstractContributionComposite.java

```diff
@@ -59,19 +61,24 @@ private <T> void initContributionsMap(Object o, IInternalExtensionRegistry exten
     }
     List<?> contributionsForMe = extensionRegistry.createContributionsFor(owner, type);
     if (type == null) {
-      m_contributionsByType = new HashMap<>();
-      m_contributionsByClass = new HashMap<>(contributionsForMe.size());
+      m_contributionsByType = null;
+      m_contributionsByClass = null;
     }
     if (CollectionUtility.hasElements(contributionsForMe)) {
+      m_contributionsByClass = new HashMap<>();
```

Comment:
```suggestion
      m_contributionsByClass = new HashMap<>(contributionsForMe.size());
```

I would also initialize the `m_contributionsByType` here as well, because the fact of having contributions suggests that they are requested later on, using `getContributionsByClass`.